controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -559,7 +559,7 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, controlPl
 	workloadCluster, err := controlPlane.GetWorkloadCluster(ctx)
 	if err != nil {
 		log.V(2).Info("cannot get remote client to workload cluster, will requeue", "cause", err)
```

Comment:
Same here, let's just drop the log and return the error directly (and use an empty result)